spec/models/conversion_host_spec.rb

```diff
@@ -6,8 +6,8 @@
   context "provider independent methods" do
     let(:host) { FactoryBot.create(:host) }
     let(:vm) { FactoryBot.create(:vm_or_template) }
-    let(:conversion_host_1) { FactoryBot.create(:conversion_host, :resource => host) }
-    let(:conversion_host_2) { FactoryBot.create(:conversion_host, :resource => vm) }
+    let(:conversion_host_1) { FactoryBot.create(:conversion_host, :skip_validate, :resource => host) }
```

Comment:
how about creating the following `vm` factory and so that you don't need to `skip_validate`?
```
factory(:vm_support_conversion, :parent => :vm) do
    after(:build) do |vm|
      vm.stub(:supports_conversion_host?).and_return true
    end
  end
``` 
and similary to the host?

Reply:
@jameswnl Seems using rspec-mocks inside factories no longer works. Take a look at the comments for  spec/factories/dialog.rb, for example.

Reply:
yeah, so this works
```
factory(:vm_support_conversion, :parent => :vm) do
    after(:build) do |obj|
      def obj.supports_conversion_host?; true; end
    end
end
```

Reply:
I think this is frowned upon, though. The only other factories that do this are tagged with TODO comments to fix them.

Reply:
ok, we can do a simple re-arrangement. After moving up the factory lines outside of context, use `allow` in `before do` as in [here](https://gist.github.com/jameswnl/c9ed7c600b1ebdac106de0fb2492ee7e#file-spec-rb-L5-L11)

Reply:
How about using a `host_redhat` factory?  That should already `supports_conversion_host?`

Reply:
I did that originally, but @jameswnl complained since it's supposed to be provider independent methods.

Reply:
I see `vm_openstack` above and `host_redhat` below so while I completely agree with that we don't need to solve that problem right here.

We should probably define a `vm_supports_conversion_host` factory in the future to handle this.

Reply:
> ok, we can do a simple re-arrangement. After moving up the factory lines outside of context, use `allow` in `before do` as in [here](https://gist.github.com/jameswnl/c9ed7c600b1ebdac106de0fb2492ee7e#file-conversion_host_spec-rb-L5-L11)

@agrare no need for the factory, just mock the method would already do

Reply:
Too late, I already switched them back. :)

Note that I also had to add an ems since the redhat provider demands a minimum api_version.